mlton/backend/rep-type.fun

```diff
@@ -449,12 +451,13 @@ structure ObjectType =
 
       (* Order in the following vector matters.  The basic pointer tycons must
        * correspond to the constants in gc/object.h.
+       * INTINF_VECTOR_TYPE_INDEX,
```

Comment:
The comment and the code should correspond.  Either do the code in alphabetical order, or do the comment in the same order as the code.

Reply:
I believe they didn't correspond in the first place, in particular the WordVector entries in the code had 32 and 16 switched from the alphabetical order locations, whereas in the comment everything was alphabetical.  I'll assume we should change this and modify this file and the corresponding C file so that everything is in alphabetical order.